### PR TITLE
Functional tests

### PR DESCRIPTION
**Pull request by dawagner**
Take over @AdrienMQ's pull request.

### CODE REVIEW

**Comment by dawagner** [comment]
@krocard @clero please review

File: CMakeLists.txt [hunk 1]
```diff
@@ -38,9 +38,12 @@ project(parameter-framework)
 
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror -Wall -Wextra")
 
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+
 option(COVERAGE "Build with coverage support" OFF)
 if(COVERAGE)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fprofile-arcs -ftest-coverage")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g -O0 -fprofile-arcs -ftest-coverage")
```
Comment (krocard) [inline CMakeLists.txt]: why ?, this is already done in DEBUG build. If you want debug info, use a debug build!

File: test/functional-tests/PfwTestCase/Types/tSTRING_128.py [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+        | Expected result :                                          |
+        | - Not Determined now                                       |
+        |============================================================|
+        """
+        log.D(self.test_Punctuation_Empty_Parenthese_String_Case.__doc__)
+        value = "ParentheseTest()"
+        log.I("STR_CHAR128 parameter Parenthese Char = %s" % (value))
+        param_check = commands.getoutput('cat $PFW_RESULT/STR_CHAR128')
+        #Set parameter value
+        out, err = self.pfw.sendCmd("setParameter", self.param_name, "'%s'" % (value))
```
Comment (krocard) [inline test/functional-tests/PfwTestCase/Types/tSTRING_128.py]: Could use str(value), would be cleaner than "%s" % value

File: test/functional-tests/PfwTestCase/Types/tSTRING_128.py [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+        | Expected result :                                          |
+        | - Not Determined now                                       |
+        |============================================================|
+        """
+        log.D(self.test_Punctuation_Full_Parenthese_String_Case.__doc__)
+        value = "ParentheseTest(test)"
+        log.I("STR_CHAR128 parameter Parenthese Char = %s" % (value))
+        param_check = commands.getoutput('cat $PFW_RESULT/STR_CHAR128')
+        #Set parameter value
+        out, err = self.pfw.sendCmd("setParameter", self.param_name, "'%s'" % value)
```
Comment (krocard) [inline test/functional-tests/PfwTestCase/Types/tSTRING_128.py]: idem

File: test/functional-tests/PfwTestCase/Types/tSTRING_128.py (hunk 2 above)
Comment (dawagner) [inline test/functional-tests/PfwTestCase/Types/tSTRING_128.py]: read again: it's not `"%s"` but `"'%s'"`

File: .gitignore [hunk 4]
```diff
@@ -1,4 +1,5 @@
-
+*.pyc
+*.pyo
```
Comment (dawagner) [inline .gitignore]: Not during compilation but during test execution. I agree that it shouldn't impact the sources. However, I see only 3 ways of doing that:

1- copy the whole test suite to the build tree
2- adding "-B" to the python command in order for python not to generate any bytecode
3- ignoring .pyc and .pyo

That 3rd solution is the most convenient.

File: CMakeLists.txt (hunk 1 above)
Comment (dawagner) [inline CMakeLists.txt]: removed.

File: .gitignore (hunk 4 above)
Comment (krocard) [inline .gitignore]: It is much better in python3.2, it generates all .py[co]  in a __pycache__ folder. I guess ignoring pyc is not so much an issue as impacts only tests. Nevertheless it still bothers me to have files generated in the sources during compilation. Anyway, this test suite is ugly, making it a little bit more isn't a big deal. :+1:

File: test/functional-tests/PfwTestCase/Types/tSTRING_128.py (hunk 2 above)
Comment (krocard) [inline test/functional-tests/PfwTestCase/Types/tSTRING_128.py]: You are right.

**Pull request merged**